### PR TITLE
fix: handle edge case when all withdrawal requests are finalized

### PR DESCRIPTION
**Pull request by bulbozaur**
## Summary
  - Fix solvency simulation test failure when `lastUnstETHIdToFinalize` exceeds `lastRequestId`
  - Add cap in `performRebase()` to limit `lastUnstETHIdToFinalize` to `lastRequestId`
  - Update `_reportAndRebase()` to handle the case when all requests are already finalized
  - Fix upper bound balance estimation by adding 100 wei margin for wstETH lock/unlock shares rounding error
  - Fix rebase calculation to account for external ether (stVaults) in fee share distribution

### CODE REVIEW

File: test/regressions/escrow-operations.t.sol [hunk 1]
```diff
@@ -369,8 +368,7 @@ contract EscrowOperationsRegressionTest is DGRegressionTestSetup {
                             .getPooledEthByShares(wstETHAmount + _initialLockedShares + _initialLockedUnStETHShares),
                         denominator: _lido.stETH.totalSupply()
                     }).toUint256(),
-                // TODO: temporarily increased delta to 10 * ACCURACY to fix possible rounding error. Need to fix it properly in a separate PR.
-                10 * ACCURACY
+                2 wei
```
Comment (Psirex) [inline test/regressions/escrow-operations.t.sol]: Consider using `ACCURACY` variable instead of `2 wei` literal

**Review by Psirex — APPROVED** [review]
